actionview/lib/action_view/record_identifier.rb

```diff
@@ -86,11 +87,38 @@ def dom_class(record_or_class, prefix = nil)
     #
     #   dom_id(Post.find(45), :edit) # => "edit_post_45"
     #   dom_id(Post.new, :custom)    # => "custom_post"
-    def dom_id(record, prefix = nil)
-      if record_id = record_key_for_dom_id(record)
-        "#{dom_class(record, prefix)}#{JOIN}#{record_id}"
+    #
+    # If you need to prepend multiple prefixes at a time, pass them as an Array
+    # or as arguments:
+    #
+    #   dom_id(Post.find(45), :edit, custom: true)                # => "edit_post_45 custom_post_45"
+    #   dom_id(Post.new, :edit, custom: true)                     # => "edit_post custom_post"
+    #
+    #   dom_id(Post.find(45), [:edit, { custom: true }, :title])  # => "edit_post_45 custom_post_45 title_post_45"
+    #   dom_id(Post.new, [:edit, { custom: true }, :title])       # => "edit_post custom_post title_post"
+    #
+    # For example, reference multiple elements from within an aria-labelledby
+    # attribute:
+    #
+    #   tag.button "Edit", aria: { labelledby: dom_id(@post, [:edit, { custom: true }, :title] }
+    #   # => <button aria-labelledby="edit_post_1 custom_post_1 title_post_1">Edit</button>
+    #
+    def dom_id(record, *prefixes, **conditional_prefixes)
+      prefixes = (Array.wrap(prefixes) + [conditional_prefixes].compact_blank).flatten
+
+      if prefixes.any?
+        identifiers = prefixes.map do |prefix|
+          case prefix
+          when Hash
+            prefix.transform_keys { |key| _dom_id(record, key) }
+          else
+            _dom_id(record, prefix.presence)
+          end
+        end
+
+        ActionView::Helpers::TagHelper.build_tag_values(identifiers).join(LIST)
```

Comment:
Ideally, this would invoke [`ActionView::TagHelper#token_list`](https://github.com/rails/rails/blob/371b4800266cbf1c659a9068f58412cb7a88e244/actionview/lib/action_view/helpers/tag_helper.rb#L327-L331) and be done with it (and not declare the `LIST` constant), but the module barriers prevent that. Is there an easy way to do that?